WcaOnRails/app/views/persons/_regional_records.html.erb

```diff
@@ -0,0 +1,50 @@
+<% results = @results.select do |result|
+     ([result.regionalSingleRecord, result.regionalAverageRecord] & record_types).any?
+   end %>
+
+<% if results.any? %>
+  <div class="records">
+    <h3 class="text-center"><%= title %></h3>
+    <div class="table-responsive">
+      <table class="table table-striped">
+        <thead>
+          <tr>
+            <th class="event"><%= t 'competitions.results_table.event' %></th>
+            <th class="single"><%= t 'common.single' %></th>
+            <th class="average"><%= t 'common.average' %></th>
+            <th class="competition"><%= t 'persons.show.competition' %></th>
+            <th class="round"><%= t 'competitions.results_table.round' %></th>
+            <th class="solves" colspan="5"><%= t 'common.solves' %></th>
+          </tr>
+        </thead>
+        <tbody>
+          <% results.group_by(&:event).each do |event, results| %>
```

Comment:
Although it's not stated directly in the docs I found this official [ruby spec](https://github.com/ruby/spec/blob/master/core/enumerable/group_by_spec.rb#L5-L10) and it definitely relies on preserving the order of both keys and values =)